### DISCUSSION TITLE
S3 gateway and TTL

### DISCUSSION THREAD

**Discussion by OffbeatPlanet** [seaweedfs/seaweedfs#2199] [Q&A] [answered]
Hi,

I've been testing SeaweedFS for a few days now, it's really a very cool piece of software! Thanks for that @chrislusf !

My use case is caching some files. For that I need to set-up TTL (all the files will have the same TTL). As other components of my software already use AWS S3, I decided to communicate with Seaweed via S3 gateway.

So my question is: is it possible to configure Seaweed to have TTL while accessing it via S3 gateway? If not, how should I do that?

Thanks a lot!

**Comment by chrislusf** [comment, marked as answer]
You can use fs.configure in weed shell to configure path specific settings.

**Reply by OffbeatPlanet** [reply]
Hi @chrislusf ,

Thanks for the very quick answer, and for sending me in that direction. I had seen the page https://github.com/chrislusf/seaweedfs/wiki/Path-Specific-Configuration but I had misunderstood one point: I thought that the filer.conf file should be created on the actual filesystem, no on Seaweed itself, so I had created that file by hand.

Creating it using `fs.configure` worked perfectly, though. I had to remove all the files of the bucket that existed before the setting the TTL up, but it seems to be a common behavior.

Hope you don't mind, but I edited the page https://github.com/chrislusf/seaweedfs/wiki/weed-shell with the output of the last version of `weed` and the `help` command. 

Thanks again for SeaweedFS!

**Reply by chrislusf** [reply]
thanks for the update! You are also welcome to improve the "path-specific-configuration" wiki page. It is a powerful feature and I hope more people can realize this. You can adjust the wiki with your point of view, and add some examples.
